testing/kuttl/e2e/exporter-standby/04-assert.yaml

```
@@ -0,0 +1,38 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+commands:
+# Grab the exporter pod
+# Check that the postgres_exporter pid is running
+# Store the exporter pid as an annotation on the pod
+- script: |
+    retry() { bash -ceu 'printf "$1\nSleeping...\n" && sleep 5' - "$@"; }
+    contains() { bash -ceu '[[ "$1" == *"$2"* ]]' - "$@"; }
+    check_containers_ready() { bash -ceu ' echo "$1" | jq -e ".[] | select(.type==\"ContainersReady\") | .status==\"True\""' - "$@";}
+
+    pod=$(kubectl get pods -o name -n $NAMESPACE \
+      -l postgres-operator.crunchydata.com/cluster=standby-cluster,postgres-operator.crunchydata.com/crunchy-postgres-exporter=true)
+    [ "$pod" = "" ] && retry "Pod not found" && exit 1
+
+    oldPid=$(kubectl get ${pod} -n ${NAMESPACE} -o jsonpath="{.metadata.annotations.oldpid}")
+    newPid=$(kubectl exec ${pod} -n ${NAMESPACE} -c exporter -- cat /tmp/postgres_exporter.pid)
+    [ "${oldPid}" -eq "${newPid}" ] && retry "pid should have changed" && exit 1
+
+    password=$(kubectl exec -n ${NAMESPACE} ${pod} -c exporter -- bash -c 'cat /opt/crunchy/password')
+    { contains "${password}" "password"; } || {
+      retry "unexpected password: ${password}"
+      exit 1
+    }
+
+    condition_json=$(kubectl get ${pod} -n ${NAMESPACE} -o jsonpath="{.status.conditions}")
+    [ "$condition_json" = "" ] && retry "conditions not found" && exit 1
+    { check_containers_ready "$condition_json"; } || {
+      retry "containers not ready"
+      exit 1
+    }
+
+    scrape_metrics=$(kubectl exec ${pod} -c exporter -n ${NAMESPACE} -- \
+      curl --silent http://localhost:9187/metrics | grep "pg_exporter_last_scrape_error")
+    { contains "${scrape_metrics}" 'pg_exporter_last_scrape_error 0'; } || {
+      retry "${scrape_metrics}"
+      exit 1
+    }
```

```suggestion
    }

```
Wonder if we can add a linter or a git pre-commit hook? (though to be fair, I think I might be the person who most often loses newlines at the end of files.)